redis/src/cluster_async/mod.rs

```diff
@@ -122,6 +128,107 @@ where
             })
     }
 
+    // Special handeling for 'SCAN' command, using cluster_scan
+    /// Perform a SCAN command on a Redis cluster, using scan state object in order to handle changes in topology
+    /// and make sure that all cluster is scanned.
+    ///
+    /// # Arguments
+    ///
+    /// * `scan_state_cursor` - A reference to the scan state, For initiating new scan send ScanStateCursor::new(),
+    /// for each iteration after the first use the returned RefCell.
+    /// * `match_pattern` - An optional match pattern of wanted keys.
+    /// * `count` - An optional count of keys wanted,
+    /// the amount returned can vary and not obligated to return exactlly count.
```

Comment:
not sure this is needed if you include the change above

Reply:
It is important to say that if you ask for 100 you might get 10, and if you ask for 10 you might get 100.
this param is just a hint to the server for how much steps to take for each scan.